Marlin/src/inc/SanityCheck.h

```diff
@@ -3173,5 +3173,89 @@ static_assert(   _ARR_TEST(3,0) && _ARR_TEST(3,1) && _ARR_TEST(3,2)
   #endif
 #endif
 
+/**
+ * Sanity check for valid stepper driver types
+ */
+#ifdef X_DRIVER_TYPE
```

Comment:
If you redefine “known” to include unspecified, then you could move these defines checks into the shared macro.